go/src/meguca/db/bans.go

```diff
@@ -121,52 +127,54 @@ func Ban(board, reason, by string, expires time.Time, log bool, ids ...uint64) (
 				sq.Update("posts").
 					Set("banned", true).
 					Where("id = ?", post.id),
-			).
-				Exec()
+			).Exec()
+
 			if err != nil {
 				return
 			}
+
 			err = bumpThread(tx, post.op, false)
+
 			if err != nil {
 				return
 			}
-			err = writeBan(tx, post.ip, board, reason, by, post.id, expires,
-				log)
+
+			e, err := writeBan(tx, post.ip, board, reason, by, post.id, expires, log)
+			
 			if err != nil {
-				return
+				return err
 			}
+
+			modLog = append(modLog, e)
 		}
+
 		return
 	})
+
 	if err != nil {
 		return
 	}
 
 	if !IsTest {
-		for _, post := range posts {
-			mLog, err := GetPostModLog(post.id)
-
-			if err != nil {
-				return err
-			}
-
-			err = auth.ModLogPost(post.id, post.op, mLog)
-
-			if err != nil {
-				return err
-			}
-
-			err = common.BanPost(post.id, post.op)
-			if err != nil {
-				return err
+		if (len(modLog) == len(posts)) {
```

Comment:
Should I have it return an error in an else block if this is not true?
Shouldn't this always be true?
Should I remove this check, then?